File: app/src/main/java/com/example/myanimedata/api/CharacterDetail.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class CharacterDetail {
    @SerializedName("mal_id")
    private final int id;

    @SerializedName("url")
    private final String url;

    @SerializedName("name")
    private final String name;

    @SerializedName("name_kanji")
    private final String nameKanji;

    @SerializedName("favorites")
    private final int favorites;

    @SerializedName("about")
    private final String about;

    @SerializedName("images")
    private final ImageResult imageResult;

    @SerializedName("anime")
    private final List<RoleResult> roleResults = new ArrayList<>();

    public int getId() {
        return id;
    }

    public String getUrl() {
        return url;
    }

    public String getName() {
        return name;
    }

    public String getNameKanji() {
        return nameKanji;
    }

    public int getFavorites() {
        return favorites;
    }

    public String getAbout() {
        return about;
    }

    public ImageResult getImageResult() {
        return imageResult;
    }

    public List<RoleResult> getRoleResults() {
        return roleResults;
    }

    public CharacterDetail(int id, String url, String name, String nameKanji, int favorites, String about, ImageResult imageResult) {
        this.id = id;
        this.url = url;
        this.name = name;
        this.nameKanji = nameKanji;
        this.favorites = favorites;
        this.about = about;
        this.imageResult = imageResult;
    }
}

File: app/src/main/java/com/example/myanimedata/api/CharacterDetail.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
class CharacterDetail(
    @field:SerializedName("mal_id") val id: Int? = 0,
    @field:SerializedName("url") val url: String? = null,
    @field:SerializedName("name") val name: String? = null,
    @field:SerializedName("favorites") val favorites: Int? = 0,
    @field:SerializedName("about") val about: String? = null,
    @field:SerializedName("images") val imageResult: ImageResult? = null,
    @field:SerializedName("anime") val roleResults: ArrayList<RoleResult>? = null
) : Parcelable